doc/examples/yy_dev/plot_max_tree.py

```diff
@@ -0,0 +1,290 @@
+"""
+========
+Max-tree
+========
+
+The max-tree is a hierarchical representation of an image that is the basis
+for a large family of morphological filters.
+
+If we apply a threshold operation to an image, we obtain a binary image
+containing one or several connected components. If we apply a lower
+threshold, all the connected components from the higher threshold are
+contained in the connected components from the lower threshold. This
+naturally defines a hierarchy of nested components that can be
+represented by a tree. whenever a connected component A obtained by
+thresholding with threshold t1 is contained in a component B obtained by
+thresholding with threshold t1 < t2, we say that B is the parent of A.
+The resulting tree structure is called a component tree. The max-tree
+is a compact representation of such a component tree. [1]_, [2]_, [3]_, [4]_
+
+In this example we give an intuition of what a max-tree is.
+
+References
+----------
+.. [1] Salembier, P., Oliveras, A., & Garrido, L. (1998). Antiextensive
+       Connected Operators for Image and Sequence Processing.
+       IEEE Transactions on Image Processing, 7(4), 555-570.
+        :DOI:`10.1109/83.663500`
+.. [2] Berger, C., Geraud, T., Levillain, R., Widynski, N., Baillard, A.,
+       Bertin, E. (2007). Effective Component Tree Computation with
+       Application to Pattern Recognition in Astronomical Imaging.
+       In International Conference on Image Processing (ICIP) (pp. 41-44).
+        :DOI:`10.1109/ICIP.2007.4379949`
+.. [3] Najman, L., & Couprie, M. (2006). Building the component tree in
+       quasi-linear time. IEEE Transactions on Image Processing, 15(11),
+       3531-3539.
+        :DOI:`10.1109/TIP.2006.877518`
+.. [4] Carlinet, E., & Geraud, T. (2014). A Comparative Review of
+       Component Tree Computation Algorithms. IEEE Transactions on Image
+       Processing, 23(9), 3885-3895.
+        :DOI:`10.1109/TIP.2014.2336551`
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.lines import Line2D
+from skimage.morphology import max_tree
+import networkx as nx
+from matplotlib.gridspec import GridSpec
+
+
+#####################################################################
+# Before we start : a few helper functions
+
+# plot_img is a helper function to plot the images and overlay the image
+# values or indices.
+def plot_img(image, ax, title, plot_text, image_values):
```

Comment:
I suggest putting `ax` first so that it matches the signature of the `Axes` methods.

Reply:
OK (I trust you with that).